sklearn/utils/_array_api.py

```diff
@@ -715,7 +728,30 @@ def _asarray_with_order(array, dtype=None, order=None, copy=None, *, xp=None):
         # container that is consistent with the input's namespace.
         return xp.asarray(array)
     else:
-        return xp.asarray(array, dtype=dtype, copy=copy)
+        return xp.asarray(array, dtype=dtype, copy=copy, device=device)
+
+
+def _ravel(array, xp=None):
+    """Array API compliant version of np.ravel.
+
+    For non numpy namespaces, it just returns a flattened array, that might
+    be or not be a copy.
+    """
+    xp, _ = get_namespace(array, xp=xp)
+    if _is_numpy_namespace(xp):
+        array = numpy.asarray(array)
+        return xp.asarray(numpy.ravel(array, order="C"))
+
+    return xp.reshape(array, shape=(-1,))
+
+
+def _item(x, xp=None):
+    """Returns the one-sized tensor as a Python number with metadata."""
```

Comment:
It does not return a Python scalar type, it returns a scalar array type from the same namespace.


```suggestion
    """Convert a 1d array with a single element to a scalar (0d) array.

    This makes it possible to preserve the dtype and device information
    of the source array(s) in scalar fitted attributes in particular.
    """
```

Reply:
I removed that entirely in latest commits